tests/rbd/rbd_utils.py

```diff
@@ -1158,3 +1160,150 @@ def verify_image_size(rbd, pool, image, size):
         return 0
     log.error(f"{image} size is not equal to expected size as {size}")
     return 1
+
+
+def resize_parent_and_clone(rbd, resize_sequence, parent_clone_spec):
+    """
+    Resize parent and clone images to compensate overhead due to the header size.
+
+    Scenarios and corresponding steps
+    1) Parent and clone are luks1 encrypted
+        Resize parent at beginning to compensate for header size,
+        clone resize not required
+    2) Parent luks1/luks2 and clone is not encrypted
+        Resize parent at beginning to compensate for header size,
+        clone resize not required since it has no header
+    3) Parent luks2 and clone luks1
+        Resize parent at beginning to compensate for header size,
+        resize clone with --allow-shrink since luks2 header is bigger than luks1
+    4) Parent not encrypted and clone is luks1/luks2
+        Resize not required before clone since parent is not encrypted, resize
+        clone with --allow-shrink post cloning to compensate for header
+    5) Parent luks1 and clone luks2
+        Resize parent before clone, resize both parent and
+        clone with --allow-shrink post cloning
+
+    Justification:
+    Since LUKS2 header is usually bigger than LUKS1 header, the rbd resize command at
+    the beginning temporarily grows the parent image to reserve some extra space in the
+    parent snapshot and consequently the cloned image. This is necessary to make all parent
+    data accessible in the cloned image. The rbd resize command at the end shrinks the
+    parent image back to its original size and does not impact the parent snapshot and
+    the cloned image to get rid of the unused reserved space
+
+    The same applies to creating a formatted clone of an unformatted image,
+    since an unformatted image does not have a header at all.
+    Args:
+        rbd: rbd object
+        resize_sequence: before_clone if resize is being performed before cloning, else after_clone
+        parent_clone_spec:
+            Example: {
+                "parent_encryption_type": "luks1",
+                "parent_spec": "pool/image",
+                "parent_passphrase": "passphrase.bin",
+                "clone_encryption_type": "luks2",
+                "clone_spec": "pool/clone",
+                "clone_passphrase": "clone_passphrase.bin",
+                "size": <image_size>
+            }
+    """
+    log.info(f"Performing resize operation {resize_sequence}")
```

Comment:
[Optional] [Minor]
Can we provide context on resizing + encryption?
This applies to first line of the method doc also

Reply:
Please let me know if the current doc and log looks okay?